blocks/procedures.js

```diff
@@ -469,6 +468,15 @@ Blockly.Blocks['procedures_mutatorarg'] = {
    */
   init: function() {
     var field = new Blockly.FieldTextInput('x', this.validator_);
+    // Hack: override showEditor to do just a little bit more work.
+    // We don't have a good place to hook into the start of a text edit.
+    var oldShowEditorFn = field.showEditor_.bind(field);
+    var newShowEditorFn = function() {
+      this.createdVariables_ = [];
+      oldShowEditorFn();
+    };
+    field.showEditor_ = newShowEditorFn.bind(field);
```

Comment:
Nit: Is the bind necessary?  Is there any chance it will be called without the `field` in `field.showEditor_()`?
Same could be said of oldShowEditorFn, as long as it is called via `field.oldShowEditorFn()` in the function expression.

Reply:
Done.